components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -99,12 +99,26 @@
 CLP_QUEUE_PASS_ENV_VAR_NAME = "CLP_QUEUE_PASS"
 CLP_REDIS_PASS_ENV_VAR_NAME = "CLP_REDIS_PASS"
 
+# Generic types
+NonEmptyStr = Annotated[str, Field(min_length=1)]
+PositiveFloat = Annotated[float, Field(gt=0)]
+PositiveInt = Annotated[int, Field(gt=0)]
+# Specific types
+Host = NonEmptyStr
+Port = Annotated[int, Field(gt=0, lt=2**16)]
+ZstdCompressionLevel = Annotated[int, Field(ge=1, le=19)]
+
 
 class StorageEngine(KebabCaseStrEnum):
     CLP = auto()
     CLP_S = auto()
 
 
+class DatabaseEngine(KebabCaseStrEnum):
+    MARIADB = auto()
+    MYSQL = auto()
```

Comment:
i can't be sure if it's a good idea to write `MARIA_DB` and `MY_SQL` so i didn't